lib/modules/manager/kustomize/artifacts.ts

```diff
@@ -0,0 +1,254 @@
+import is from '@sindresorhus/is';
+import { quote } from 'shlex';
+import upath from 'upath';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type {
+  ExecOptions,
+  ExtraEnv,
+  ToolConstraint,
+} from '../../../util/exec/types';
+import {
+  deleteLocalFile,
+  getSiblingFileName,
+  localPathExists,
+  privateCacheDir,
+  readLocalFile,
+} from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import { DockerDatasource } from '../../datasource/docker';
+import { HelmDatasource } from '../../datasource/helm';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import { parseKustomize } from './extract';
+
+function generateHelmEnvs(): ExtraEnv {
+  return {
```

Comment:
```suggestion
  const cacheDir = privateCacheDir();
  return {
```

don't call this multiple times (perf)